content/blog/2020-05-googlesheets4-0-2-0/index.Rmarkdown.orig

```diff
@@ -0,0 +1,186 @@
+
+*Why **4**? Why googlesheets**4**? Did I miss googlesheets1 through 3? No. The idea is to name the package after the corresponding version of the Sheets API. In hindsight, the original googlesheets should have been googlesheets**3**.*
+
+googlesheets4 wraps [v4 of the Sheets API](https://developers.google.com/sheets), whereas the original googlesheets wraps v3. The [v3 API](https://developers.google.com/sheets/api/v3) has been deprecated for a long time. Its shut-off has been postponed several times, but its days are clearly numbered. Gradual shutdown of certain endpoints and scopes is already underway, with complete shutdown [expected on September 30, 2020](https://cloud.google.com/blog/products/g-suite/migrate-your-apps-use-latest-sheets-api) at the time of writing. If you use googlesheets, you **must** switch to googlesheets4 and the time is now.
+
+Change is hard, but let's focus on the positive:
+
+* The v4 API has a better design, which makes it easier to wrap, which
+  translates to a better user experience in client packages.
+* I have gotten a lot better at writing R packages. Some of the fiddly parts
+  of the original googlesheets came from the awkward v3 API, but some of it
+  came from me.
+* The gargle package ([gargle.r-lib.org](https://gargle.r-lib.org)) provides
+  infrastructure common to ~250 Google APIs, including Sheets, Drive, Gmail, and
+  BigQuery. By using gargle for auth, we can provide a consistent auth
+  experience across googlesheets4, googledrive, gmailr, and bigrquery.
+* The googledrive package ([googledrive.tidyverse.org](https://googledrive.tidyverse.org))
+  provides a full-featured client for working with files on Google Drive, which
+  allows googlesheets4 to focus on operations specific to spreadsheets. In the
+  original googlesheets package, about half of the code and effort was actually
+  devoted to Drive, not Sheets.
+
+## Read a Sheet
+
+Let's say you're staring at a Sheet in the browser and you want to read it into R. Copy the URL to your clipboard and paste it into a call to `read_sheet()` like this:
+
+```{r}
+read_sheet("https://docs.google.com/spreadsheets/d/1U6Cf_qEOhiR9AZqTqS3mbMF3zt2db48ZP5v3rkrAEJY/edit#gid=780868077")
+```
+
+I'm reading from one of our public example Sheets -- specifically, a Sheet that holds Gapminder data. `gs4_examples()` and `gs4_example()` make the example Sheets easy to access.
+
+*If you're following along at home, you probably just got a prompt to log in with Google. That's because, in general, you'll want googlesheets4 to be able to do the same things you can do with Sheets in the browser. If you know you only want to read public Sheets, you can use `gs4_deauth()` to tell googlesheets4 that it should not attempt auth.*
+
+```{r}
+gs4_example("gapminder")
+```
+
+The above demonstrates that printing a Sheet ID reveals relevant metadata about the Sheet, such as its name and an overview of its worksheets and named ranges.
```

Comment:
I think you need to say something about `gs4_example()` returning a Sheet ID and how to find ids for your own sheets? Or maybe that's not really the point of this section which is mostly about reading specified ranges from a sheet, and doing `id <- gs4_example("gapminder")` might help the flow?